src/triangle.rs

```diff
@@ -0,0 +1,356 @@
+
+    /// empty [`Triangle`] at (1.0, 1.0)
+    pub const ONE: Self = Self::from_coords((1.0, 1.0), (1.0, 1.0), (1.0, 1.0));
+
+    /// equilateral [`Triangle`]
+    pub const EQUILATERAL: Self = Self::from_coords(
+        (
+            1.0 / 2.0,
+            1.732050807568877293527446341505872367_f64 / 2.0, /* (sqrt 3)/2 */
+        ),
+        (0.0, 0.0),
+        (1.0, 0.0),
+    );
+
+    /// A new [`Triangle`] from three vertices ([`Points`])
+    #[inline]
+    pub fn new(a: impl Into<Point>, b: impl Into<Point>, c: impl Into<Point>) -> Self {
+        Self {
+            a: a.into(),
+            b: b.into(),
+            c: c.into(),
+        }
+    }
+
+    /// A new [`Triangle`] from three float vertex coordinates
+    #[inline]
+    pub const fn from_coords(a: (f64, f64), b: (f64, f64), c: (f64, f64)) -> Self {
+        Self {
+            a: Point::new(a.0, a.1),
+            b: Point::new(b.0, b.1),
+            c: Point::new(c.0, c.1),
+        }
+    }
+
+    /// A new [`Triangle`] from centroid and offsets ([`Vec2`]) - offset of each vertex from the centroid
+    #[inline]
+    fn from_centroid_offsets(centroid: impl Into<Point>, offsets: [impl Into<Vec2>; 3]) -> Self {
+        let centroid = centroid.into();
+
+        let mut points: [Point; 3] = Default::default();
+        for (i, offset) in offsets.into_iter().enumerate() {
+            points[i] = centroid + offset.into();
+        }
+
+        Self::new(points[0], points[1], points[2])
+    }
+
+    /// The centroid of the [`Triangle`]
+    #[inline]
+    pub fn centroid(&self) -> Point {
+        (1.0 / 3.0 * (self.a.to_vec2() + self.b.to_vec2() + self.c.to_vec2())).to_point()
+    }
+
+    /// The circumcenter of the [`Triangle`]
+    #[inline]
+    pub fn circumcenter(&self) -> Point {
+        let d = 2.0
+            * (self.a.x * (self.b.y - self.c.y)
+                + self.b.x * (self.c.y - self.a.y)
+                + self.c.x * (self.a.y - self.b.y));
+
+        let ux = ((self.a.x.powi(2) + self.a.y.powi(2)) * (self.b.y - self.c.y)
+            + (self.b.x.powi(2) + self.b.y.powi(2)) * (self.c.y - self.a.y)
+            + (self.c.x.powi(2) + self.c.y.powi(2)) * (self.a.y - self.b.y))
+            / d;
+
+        let uy = ((self.a.x.powi(2) + self.a.y.powi(2)) * (self.c.x - self.b.x)
+            + (self.b.x.powi(2) + self.b.y.powi(2)) * (self.a.x - self.c.x)
+            + (self.c.x.powi(2) + self.c.y.powi(2)) * (self.b.x - self.a.x))
+            / d;
+
+        Point::new(ux, uy)
+    }
+
+    /// The offset of each vertex from the centroid
+    #[inline]
+    pub fn offsets(&self) -> [Vec2; 3] {
+        let centroid = self.centroid().to_vec2();
+
+        [
+            (self.a.to_vec2() - centroid),
+            (self.b.to_vec2() - centroid),
+            (self.c.to_vec2() - centroid),
+        ]
+    }
+
+    /// The area of the [`Triangle`]
+    #[inline]
+    pub fn area(&self) -> f64 {
+        0.5 * (self.b - self.a).cross(self.c - self.a).abs()
```

Comment:
I disagree that there should be an `abs` here. We default to signed area in kurbo, and by taking the absolute value we'll have situations where the return value of this function doesn't match the area of converting into shape first.

Reply:
not exactly sure the use of this cause i dont know of any benefits for signed area but ill just do it if its the norm.

should the area test be updated to be negative now?

Reply:
Yes, or permute the order of vertices. Converting to shape and evaluating area gives a negative result, so that's confirmation that the sign conventions are consistent. And in fact, you can make a good argument that this should be tested explicitly.

Reply:
> Converting to shape and evaluating area gives a negative result, so that's confirmation that the sign conventions are consistent

but doesn't the shape implementation default to triangles pre-existing signed area function? 

no problem though, ive changed to test the original and one with a vertex permutation